test/sql/blob/test_blob.cpp

```diff
@@ -0,0 +1,268 @@
+	REQUIRE_FAIL(con.Query("INSERT INTO blobs VALUES('\\xAAAFFAA'), ('\\xAAAFFAAAAFFAA'), ('\\xAAAFFAAAAFFAAAAFFAA')"));
+
+	// insert BLOB with “non-printable” octets
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::BYTEA)"));
+
+	// insert BLOB with “non-printable” octets, but now using VARCHAR string (should fail)
+	REQUIRE_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::VARCHAR)"));
+	REQUIRE_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124')"));
+
+	// insert BLOB with “non-printable” octets, but now using string
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::BLOB)"));
+}
+
+TEST_CASE("Select BLOB values", "[blob]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE blobs (b BYTEA);"));
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\\xFF00AA'), ('a a'::BYTEA)"));
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::BYTEA)"));
+
+	result = con.Query("SELECT * FROM blobs");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BLOB("\\xFF00AA"), Value::BLOB("a a"), Value::BLOB("\153\154\155 \052\251\124")}));
+
+	//BLOB with “non-printable” octets
+	REQUIRE_NO_FAIL(con.Query("SELECT 'abc \201'::BYTEA;"));
+	result = con.Query("SELECT 'abc \201'::BYTEA;");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BLOB("abc \201")}));
+
+	REQUIRE_NO_FAIL(con.Query("SELECT 'abc \153\154\155 \052\251\124'::BYTEA;"));
+	result = con.Query("SELECT 'abc \153\154\155 \052\251\124'::BYTEA;");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BLOB("abc \153\154\155 \052\251\124")}));
+
+	//now VARCHAR with “non-printable” octets, should fail
+	REQUIRE_FAIL(con.Query("SELECT 'abc \201'::VARCHAR;"));
+	REQUIRE_FAIL(con.Query("SELECT 'abc \153\154\155 \052\251\124'::VARCHAR;"));
+}
+
+TEST_CASE("Test BLOB with PreparedStatement from a file", "[blob]") {
+	unique_ptr<QueryResult> result;
+	unique_ptr<QueryResult> result_other;
+	DuckDB db(nullptr);
+	Connection con(db);
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE blobs (b BYTEA);"));
+
+	string blob_file_path = TestCreatePath("blob_file.txt");
+    ofstream ofs_blob_file(blob_file_path, std::ofstream::out | std::ofstream::app);
+    // Insert all ASCII chars from 1 to 255, avoiding only the '\0' char
+	char ch = '\1';
+	for(idx_t i = 0; i < 255; ++i, ++ch) {
+    	ofs_blob_file << ch;
+	}
+	ofs_blob_file.close();
+
+	// Blob file with all ascii chars, except '\0'
+	ifstream ifs(blob_file_path, ifstream::binary);
+	REQUIRE(ifs.is_open());
+
+	// get pointer to associated buffer object
+	filebuf *file_buf = ifs.rdbuf();
+
+	// get file size using buffer's members
+	size_t size = file_buf->pubseekoff(0, ifs.end, ifs.in);
+	file_buf->pubseekpos (0, ifs.in);
+
+	// allocate memory to contain file data
+	unique_ptr<char[]> buffer(new char[size + 1]);
+
+	// get file data
+	file_buf->sgetn(buffer.get(), size);
```

Comment:
Correct me if I'm wrong, but it looks like the file itself is not read by the system here, right? Perhaps test a `COPY INTO` statement here? Otherwise, I would just remove the creating/reading of the file here and only test prepared statements.

Reply:
Yeah, the ifstream is reading the file, my idea was to test the insertion of BLOB values from the file system, if DuckDB can support to load data from arbitrary files such as images (.png, .jpeg, etc),  but I think that COPY INTO should be a better fit for that. Does it support binary files?

Reply:
The current implementation loads the file using a fstream, so from the perspective of DuckDB this is equivalent to loading from memory but in a complicated way.

COPY INTO only supports CSV files, but perhaps you can test if a blob works in a CSV file. It should work, as long as UTF8 validation is skipped for blobs.